### PR TITLE
[SDP-1078] Implement DistributionAccountDBSignatureClient

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Implement DistributionAccountDBSignatureClient, that allows having one distribution account per tenant and store it in the DB, in encryption form.

### Why

Closes https://stellarorg.atlassian.net/browse/SDP-1078

### Further work

This is just one implementation for Distribution accounts signing. More options will be implemented down the line, like VAULT for instance.

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [x] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: db/migrations/tss-migrations/2024-02-15.0-add-stellar-signatories-table.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+-- +migrate Up
+
+CREATE TABLE stellar_signatories (
+    public_key VARCHAR(64) PRIMARY KEY,
+    encrypted_private_key VARCHAR(256),
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
```
Comment (marcelosalloum) [inline db/migrations/tss-migrations/2024-02-15.0-add-stellar-signatories-table.sql:8]: I'm still debating if this is the best name for this table. Here's a list of names I discarded:
- **stellar_keypairs**: keypair already has a solid meaning of the pair of public/private stellar keys, and it's used inside the signature clients. Also, this table would have the private key encrypted, which is a bit different than the traditional keypair.
- **distribution_accounts**: this clearly means that this table will be used explicitly for distribution accounts, which may not be true in the mid-term. For instance, we could decide to store all private keys here, or at least store the host private key alongside the other distribution keys. For that reason, I wanted to use a more generic term.

The other option I have in mind is stellar_credentials, I'm keen to hear from others what they think would work best

Comment (marwen-abid) [reply db/migrations/tss-migrations/2024-02-15.0-add-stellar-signatories-table.sql:8]: `stellar_signatories` is a good option. 

Another one would be `stellar_key_vault` ? there are a few products named `key vault` which do just that: store cryptographic keys. 
https://azure.microsoft.com/en-ca/products/key-vault
https://www.oracle.com/ca-en/security/database-security/key-vault/

**Review by marwen-abid — APPROVED** [review]
LGTM! This is amazing 🙌

File: db/migrations/tss-migrations/2024-02-15.0-add-stellar-signatories-table.sql (hunk 1 above)
Comment (marcelosalloum) [reply db/migrations/tss-migrations/2024-02-15.0-add-stellar-signatories-table.sql]: After some discussion offline we agreed to go with `vault` for the table name, and DBVault and DBVaultEntry for the object names.

**Pull request merged**